### DISCUSSION TITLE
NetBox Upgrade to 4.X - Topology Views Plugin Not Saving Coordinates

### DISCUSSION THREAD

**Discussion by Navesx0** [mattieserver/netbox-topology-views#577] [Q&A]
Hello everyone,

I am currently performing the upgrade procedure for NetBox from version 3.6.7 to version 4.1.2. However, in the 4.x versions, it appears that the Topology Views plugin is not saving the coordinates.

We are using NetBox in a Kubernetes cluster, where the deployment is done via the NetBox Community Helm chart. The plugin installation is being carried out in the image as per the NetBox Docker documentation from the NetBox Community.

In NetBox version 3.6.7, with the plugin version 3.8.1, the functionality to save the coordinates works normally.

In version 4.X, when I enable the "Save Coordinates" option in the pod, I get the following success log:

```
"PATCH /api/plugins/netbox_topology_views/save-coords/save_coords/ HTTP/1.1" 200 25 "https://netbox-hml.test.com/plugins/netbox_topology_views/topology/?save_coords=on&show_cables=on&site_id=1&location_id=75" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36"
"GET /status/applications/netbox/processes/running HTTP/1.1" 200 3 "-" "kube-probe/1.28"
"GET /login/ HTTP/1.1" 200 4872 "-" "kube-probe/1.28"
```

However, in the 4.X versions, I don't get any log output when trying to save the coordinates. I have already tested the Topology Views plugin with versions v4.0.0-beta.2, v4.0.0, v4.0.1, and v4.1.0.

The plugin config:

```
PLUGINS = [
    'netbox_topology_views'
]

PLUGINS_CONFIG = {
    'netbox_topology_views': {
        'device_img': [
            'router', 'access-switch', 'dio', 'firewall', 'core-switch',
            'isp-cpe-material', 'server', 'pc', 'wireless-ap', 'hub', 'ata',
            'distribution-switch', 'patchpanel', 'thinclient', 'pontorede',
            'linkrede', 'linkfibra', 'medidorenergia', 'rep'
        ],
        'preselected_device_roles': '',
        'preselected_tags': '',
        'allow_coordinates_saving': True,
        'enable_circuit_terminations': True,
        'draw_default_layout': False,
    }
}
```

Note: The application does not show any errors except for not saving the Topology Views coordinates.

Thank you in advance for any help!

**Comment by dreng** [comment]
Hi,

I see no reason why it shouldn't work. I just can assure you that it works in general. So there must be something special in your environment. I personally don't use docker but that shouldn't make a difference. I assume that other things will be stored in the database sucessfully.

Have you tried to install a fresh docker container with a blank NetBox instance just to have a prove that it may work?

Just a side note: Your config seems to be quite old. Some parts have moved into the GUI, so it's not necessary to define them in the config file. This is how the config looks like in my development environment:

```
PLUGINS_CONFIG = {
    'netbox_topology_views': {
        'static_image_directory': 'netbox_topology_views/img',
        'allow_coordinates_saving': True,
#        'always_save_coordinates': True
    }
}
```

Please refer to the [docs](https://github.com/netbox-community/netbox-topology-views#individual-options).

**Reply by Navesx0** [reply]
Hello, first of all, thank you for your help.
Yes, I tested installing NetBox (v4.1.2) with a blank database, and I was able to register equipment normally. However, the plugin did not save the coordinates.
I tested the deployment with version 3.7.8, and the plugin saved the coordinates correctly.
What I am struggling with is that the deployment procedure I am using works correctly for versions 3.6.X and 3.7.X, but in version 4.X, the only issue is that the Topology Views plugin does not save the coordinates.
